src/main/scala/mesosphere/marathon/api/v2/json/AppUpdate.scala

```diff
@@ -132,7 +125,8 @@ object AppUpdate {
     appUp.dependencies is valid
     appUp.upgradeStrategy is valid
     appUp.storeUrls is optional(every(urlCanBeResolvedValidator))
-    appUp.ports is optional(elementsAreUnique(AppDefinition.filterOutRandomPorts))
+    appUp.ports is optional(elementsAreUnique(AppDefinition.filterOutRandomPorts, "Ports must be unique."))
+    appUp.portDefinitions is optional(PortDefinitions.portDefinitionsValidator)
```

Comment:
If you import PortDefinitions and rename the validator, it could look like:
`appUp.portDefinitions is optional(portsAreUnique)`